torch/csrc/jit/python/python_sugared_value.cpp

```diff
@@ -252,6 +301,11 @@ SugaredValuePtr ModuleValue::getitem(
         }
       }
       throw ErrorReport(loc) << "Key Error, " << idx_str;
+    } else if (hint_) {
```

Comment:
We already have `concreteType_`, can we just look it up there ?

Reply:
In the current implementation, these contained type hints live on the parent concrete type of the module to which they apply, or in the case of an annotation on `self`, in the concrete type of the module itself. In one case, the hint needs to be threaded through and in another case it doesn't. 

Given that a module can only appear once in a module hierarchy, storing these contained type hints in the concrete types of the modules to which they apply would also work and make the code cleaner, but might be more confusing to understand.